### PR TITLE
fix issue #11

### LINKED ISSUE #11: Not correctly parse incomplete process
on some process when using /service endpoint (mostly at YE BEING_EDITTED_NAMESPACE) API breaks with key error

`KeyError: &#x27;ΥΕ:Διασυνοριακή Παροχή Υπηρεσιών Μεσιτείας Ακινήτων στην Ελληνική Επικράτεια&#x27; // Werkzeug Debugger`
[response_1620849804377.txt](https://github.com/ckarageorgkaneen/diavlos/files/6468821/response_1620849804377.txt)

that process is incomplete
```
{
  "result": {
    "fields": {
      "Process": {
        "1": {
          "process_org_owner": "Φορέας:ΕΔΥΤΕ",
          "process_usage": "Δικαιολογητικό"
        }
      }
    },
    "fullname": "ΥΕ:Διασυνοριακή Παροχή Υπηρεσιών Μεσιτείας Ακινήτων στην Ελληνική Επικράτεια",
    "name": "Διασυνοριακή Παροχή Υπηρεσιών Μεσιτείας Ακινήτων στην Ελληνική Επικράτεια",
    "page_id": 15018,
    "update": "2021-04-20T07:44:03"
  },
  "success": true
}
```
the problem is that cant walk all the tree using `page_continue`  

may need somewhere on code a try: catch to handle that situation.

### PR DESCRIPTION
**Pull request by sl45sms**
Resolve #11 It is a KeyError, the key is missing from the incomplete process. So catch the error.

### CODE REVIEW

File: diavlos/src/service/service.py [hunk 1]
```diff
@@ -127,7 +127,10 @@ def _id_by_fullname(self,name_):
         else:
             site_response_results = site_response['query']['results']
             if len(site_response_results) >= 1:
-                result = site_response['query']['results'][name_]['printouts']['process_id']
```
Comment (ckarageorgkaneen) [inline diavlos/src/service/service.py:130]: Better do:

```
if len(site_response_results) >= 1 and name_ in site_response_results:
     result = site_response_results[name_]['printouts']['process_id']
else:
     result = None
```

File: diavlos/src/service/service.py [hunk 2]
```diff
@@ -260,8 +263,8 @@ def fetch_by_name(self, name, fetch_bpmn_digital_steps=None):
                     service_dict).replace('\n', '').replace(
                     '\t', '').replace('\"', '\'')
 
-            # data = {**data, **{"update":latest_update_date,"page_id":page_id}}
-            # data = {'data': data, "update":latest_update_date,"page_id":page_id}
+            data = {**data, **{"update":latest_update_date,"page_id":page_id}}
```
Comment (ckarageorgkaneen) [inline diavlos/src/service/service.py:266]: data variable seems redundant. Just do: `result = {...}`

**Review by ckarageorgkaneen — CHANGES_REQUESTED** [review]

File: diavlos/src/service/service.py (hunk 1 above)
Comment (sl45sms) [reply diavlos/src/service/service.py:130]: Yes, that is more elegant.

File: diavlos/src/service/service.py (hunk 2 above)
Comment (sl45sms) [reply diavlos/src/service/service.py]: I've refactored this to fix also an issue that Breaks  BPMN #10

**Review by ckarageorgkaneen — CHANGES_REQUESTED** [review]

**Comment by sl45sms** [comment]
I've made a mess with this one, I will send another pull request with changes.

**Pull request closed without merging**